tools/speech_dataset_processor/sdp/utils/common.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import tarfile
+import urllib
+
+import wget
+
+from nemo.utils import logging
+
+# TODO: seems like this download code is saving initial file
+#     in the local directory!
+
+
+def download_file(source_url: str, target_directory: str):
+    logging.info(f"Trying to download data from {source_url} " f"and save it in this directory: {target_directory}")
+    filename = os.path.basename(urllib.parse.urlparse(source_url).path)
```

Comment:
Save cwd with cwd = os.getcwd(), use os.chdir(new dir) inside of a `try: ... finally: ...` block and revert back to original cwd.